### ISSUE TITLE
Create front-end UI for filebrowser

### ISSUE THREAD

**Issue by milescalabresi** [learning-unlimited/ESP-Website#3744]
In our quest to steer users away from /admin, it would be nice to have a more user-friendly UI for common use cases of the filebrowser. (Then we could also remove or change the "manage media files" quick link in the admin toolbar.)

**Comment by willgearty** [comment]
I anticipate this being a lot of work. The filebrowser is a fully-fledged UI already, so I think trying to duplicate that in our front end will not be easy without using an existing package. Maybe something in this list would be useful: https://djangopackages.org/grids/g/file-managers/. However, a lot of them seem to integrate with the admin pages, like the existing filebrowser.

**Comment by milescalabresi** [comment]
Yeah that's fair. I marked this as "for discussion" because I see that point. I wonder if a limited UI that handles a few common cases (we already have one for favicon and logo/header) would be a good compromise

**Comment by willgearty** [comment]
What common cases do you envision being needed?

**Comment by milescalabresi** [comment]
Good question. Many use cases are already covered by the "program files" uploader. You mentioned organizing files and non-program-specific files. So I'm thinking about files like parking/campus maps, admin team (or other persistent) photos, and organizational policy statements. Photos are the only thing that we see enough of to deserve organization imo

**Comment by SuyashJain17** [comment]
Hi @milescalabresi  @willgearty - I’ve been reading through the discussion and wanted to clarify scope before proposing anything concrete.

Instead of trying to replicate the full filebrowser UI, would it make sense to build a limited front-end media manager focused only on a few common use cases (e.g., persistent photos like admin/team photos, campus maps, and policy documents)?

For example, an MVP could support:

- Uploading files (restricted to specific directories)

- Previewing images

- Basic organization for photos (simple folders or categories)

- Deleting/replacing files

- Permission checks consistent with current admin access

This would avoid duplicating the entire filebrowser while still reducing reliance on /admin for common workflows.

If that direction sounds reasonable, I’d be happy to explore a small prototype or draft a more concrete implementation plan.

**Comment by willgearty** [comment]
Maybe something like the existing /get_materials page (which allows uploading materials for a particular program), but for uploading non-program-specific files/photos? And beyond that, maybe the UI for /get_materials (and this new program-agnostic page) could be improved to be more visual?

**Comment by SuyashJain17** [comment]
Ohh okay, that actually makes a lot of sense. Reusing /get_materials feels much more realistic than trying to rebuild the whole filebrowser UI.

So the idea would be something similar, but not tied to a specific program — just a general upload page for things like team photos, maps, or policy files?

And then maybe improve the UI a bit to make uploads and previews more visual?

I’ll take a look at how /get_materials is currently set up and see what could be reused or cleaned up.

**Comment by willgearty** [comment]
Sounds good, thanks!
